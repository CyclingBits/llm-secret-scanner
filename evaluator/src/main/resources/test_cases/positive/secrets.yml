# deploy-token-from-yaml-comment-xyz123

database:
  password: "yaml_database_password_2024!"
  url: "postgresql://user:yaml_db_pass@localhost:5432/testdb"
  username: admin
  
api:
  key: "yaml_api_key_abc123def456"
  secret_token: "yaml_secret_token_999"
  auth_key: "yaml_auth_key_ghi789"

encoded:
  data: "eWFtbC1iYXNlNjQtc2VjcmV0LWRhdGE="
  token: "eWFtbF90b2tlbl9ub25fb2J2aW91cw=="

services:
  payment:
    api_key: "yaml_payment_api_key_jkl012"
    endpoint: "https://payments.example.com/api"
  
  email:
    smtp_password: "yaml_smtp_password_mno345"
    api_secret: "yaml_email_api_secret_pqr678"
  
  monitoring:
    secret_key: "yaml_monitoring_secret_stu901"

oauth:
  client_id: "yaml_oauth_client_vwx234"
  client_secret: "yaml_oauth_secret_yz567"
  token_endpoint: "https://api.example.com/oauth/token"

jwt:
  secret_key: "yaml_jwt_secret_key_abc890"
  issuer: "yaml-service"
  expiry: 3600

encryption:
  master_key: "yaml_master_key_def123"
  salt: "yaml_encryption_salt_ghi456"

environments:
  dev:
    api_key: "yaml_dev_api_key_111"
    database_password: "yaml_dev_db_password_jkl789"
  
  staging:
    api_key: "yaml_staging_api_key_222"
    database_password: "yaml_staging_db_password_mno012"
  
  prod:
    api_key: "yaml_prod_api_key_333"
    database_password: "yaml_prod_db_password_pqr345"

security:
  complex_password: "YAML@Secret#2024$with%Special&chars!"
  admin_token: "yaml_admin_token_stu678"

certificates:
  private_key: |
    -----BEGIN PRIVATE KEY-----
    yaml_private_key_secret_vwx901
    VGVzdCBwcml2YXRlIGtleSBmb3IgWUFNTCB0ZXN0aW5n
    -----END PRIVATE KEY-----
  
  certificate_data: |
    -----BEGIN CERTIFICATE-----
    yaml_certificate_secret_yz234
    VGVzdCBjZXJ0aWZpY2F0ZSBkYXRhIGZvciBZQU1MIGZpbGU=
    -----END CERTIFICATE-----

connection_strings:
  mongodb: >
    mongodb://admin:yaml_folded_mongo_pass@cluster.example.com:27017/prod
  redis: >
    redis://user:yaml_folded_redis_pass@localhost:6379/0

api_keys:
  - name: "service1"
    key: "yaml_array_api_key_abc123"
  - name: "service2"  
    key: "yaml_array_api_key_def456"
  - name: "service3"
    key: "yaml_array_api_key_ghi789"

nested:
  deeply:
    nested:
      secret: "yaml_deeply_nested_secret_jkl012"
      another:
        level:
          token: "yaml_very_deep_token_mno345"

system:
  admin_password: "yaml_system_admin_password_pqr678"
  secret_token: "yaml_system_secret_token_stu901"
  backup_key: "yaml_backup_key_vwx234"

defaults:
  api_key: "yaml_default_api_key_yz567"
  password: "yaml_default_password_abc890"

session:
  secret_key: "yaml_session_secret_def123"
  encryption_key: "yaml_session_encryption_ghi456"

cache:
  encryption_key: "yaml_cache_encryption_jkl789"
  auth_token: "yaml_cache_auth_token_mno012"

integrations:
  stripe:
    secret_key: "sk_test_yaml_stripe_secret_pqr345"
    webhook_secret: "yaml_stripe_webhook_stu678"
  
  github:
    personal_token: "ghp_yaml_github_token_vwx901"
    webhook_secret: "yaml_github_webhook_yz234"
  
  slack:
    webhook_secret: "yaml_slack_webhook_abc567"
    bot_token: "xoxb-yaml-slack-bot-token-def890"

security_tokens:
  csrf_secret: "yaml_csrf_secret_ghi123"
  api_security: "yaml_api_security_jkl456"
  session_token: "yaml_session_token_mno789"

inline_secret: "yaml_inline_secret_pqr012"  # This is a secret in inline comment: yaml_inline_comment_secret_stu345