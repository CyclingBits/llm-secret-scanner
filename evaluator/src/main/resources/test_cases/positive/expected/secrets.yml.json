[
  {
    "filePath": "secrets.yml",
    "issueNumber": 1,
    "lineNumber": 1,
    "issueType": "Token",
    "secretValue": "deploy-token-from-yaml-comment-xyz123"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 2,
    "lineNumber": 4,
    "issueType": "Password",
    "secretValue": "yaml_database_password_2024!"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 3,
    "lineNumber": 5,
    "issueType": "URL Credential",
    "secretValue": "yaml_db_pass"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 4,
    "lineNumber": 9,
    "issueType": "API Key",
    "secretValue": "yaml_api_key_abc123def456"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 5,
    "lineNumber": 10,
    "issueType": "Token",
    "secretValue": "yaml_secret_token_999"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 6,
    "lineNumber": 11,
    "issueType": "Key",
    "secretValue": "yaml_auth_key_ghi789"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 7,
    "lineNumber": 14,
    "issueType": "Encoded Secret",
    "secretValue": "eWFtbC1iYXNlNjQtc2VjcmV0LWRhdGE="
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 8,
    "lineNumber": 15,
    "issueType": "Encoded Secret",
    "secretValue": "eWFtbF90b2tlbl9ub25fb2J2aW91cw=="
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 9,
    "lineNumber": 19,
    "issueType": "API Key",
    "secretValue": "yaml_payment_api_key_jkl012"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 10,
    "lineNumber": 23,
    "issueType": "Password",
    "secretValue": "yaml_smtp_password_mno345"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 11,
    "lineNumber": 24,
    "issueType": "API Secret",
    "secretValue": "yaml_email_api_secret_pqr678"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 12,
    "lineNumber": 27,
    "issueType": "Secret",
    "secretValue": "yaml_monitoring_secret_stu901"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 13,
    "lineNumber": 31,
    "issueType": "Client Secret",
    "secretValue": "yaml_oauth_secret_yz567"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 14,
    "lineNumber": 35,
    "issueType": "JWT Secret",
    "secretValue": "yaml_jwt_secret_key_abc890"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 15,
    "lineNumber": 40,
    "issueType": "Key",
    "secretValue": "yaml_master_key_def123"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 16,
    "lineNumber": 41,
    "issueType": "Salt",
    "secretValue": "yaml_encryption_salt_ghi456"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 17,
    "lineNumber": 45,
    "issueType": "API Key",
    "secretValue": "yaml_dev_api_key_111"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 18,
    "lineNumber": 46,
    "issueType": "Password",
    "secretValue": "yaml_dev_db_password_jkl789"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 19,
    "lineNumber": 49,
    "issueType": "API Key",
    "secretValue": "yaml_staging_api_key_222"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 20,
    "lineNumber": 50,
    "issueType": "Password",
    "secretValue": "yaml_staging_db_password_mno012"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 21,
    "lineNumber": 53,
    "issueType": "API Key",
    "secretValue": "yaml_prod_api_key_333"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 22,
    "lineNumber": 54,
    "issueType": "Password",
    "secretValue": "yaml_prod_db_password_pqr345"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 23,
    "lineNumber": 57,
    "issueType": "Password",
    "secretValue": "YAML@Secret#2024$with%Special&chars!"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 24,
    "lineNumber": 58,
    "issueType": "Token",
    "secretValue": "yaml_admin_token_stu678"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 25,
    "lineNumber": 62,
    "issueType": "Private Key",
    "secretValue": "-----BEGIN PRIVATE KEY-----"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 26,
    "lineNumber": 68,
    "issueType": "Certificate",
    "secretValue": "-----BEGIN CERTIFICATE-----"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 27,
    "lineNumber": 75,
    "issueType": "URL Credential",
    "secretValue": "yaml_folded_mongo_pass"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 28,
    "lineNumber": 77,
    "issueType": "URL Credential",
    "secretValue": "yaml_folded_redis_pass"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 29,
    "lineNumber": 81,
    "issueType": "API Key",
    "secretValue": "yaml_array_api_key_abc123"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 30,
    "lineNumber": 83,
    "issueType": "API Key",
    "secretValue": "yaml_array_api_key_def456"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 31,
    "lineNumber": 85,
    "issueType": "API Key",
    "secretValue": "yaml_array_api_key_ghi789"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 32,
    "lineNumber": 90,
    "issueType": "Secret",
    "secretValue": "yaml_deeply_nested_secret_jkl012"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 33,
    "lineNumber": 93,
    "issueType": "Token",
    "secretValue": "yaml_very_deep_token_mno345"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 34,
    "lineNumber": 96,
    "issueType": "Password",
    "secretValue": "yaml_system_admin_password_pqr678"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 35,
    "lineNumber": 97,
    "issueType": "Token",
    "secretValue": "yaml_system_secret_token_stu901"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 36,
    "lineNumber": 98,
    "issueType": "Key",
    "secretValue": "yaml_backup_key_vwx234"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 37,
    "lineNumber": 101,
    "issueType": "API Key",
    "secretValue": "yaml_default_api_key_yz567"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 38,
    "lineNumber": 102,
    "issueType": "Password",
    "secretValue": "yaml_default_password_abc890"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 39,
    "lineNumber": 105,
    "issueType": "Secret Key",
    "secretValue": "yaml_session_secret_def123"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 40,
    "lineNumber": 106,
    "issueType": "Key",
    "secretValue": "yaml_session_encryption_ghi456"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 41,
    "lineNumber": 109,
    "issueType": "Key",
    "secretValue": "yaml_cache_encryption_jkl789"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 42,
    "lineNumber": 110,
    "issueType": "Token",
    "secretValue": "yaml_cache_auth_token_mno012"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 43,
    "lineNumber": 114,
    "issueType": "Secret Key",
    "secretValue": "sk_test_yaml_stripe_secret_pqr345"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 44,
    "lineNumber": 115,
    "issueType": "Secret",
    "secretValue": "yaml_stripe_webhook_stu678"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 45,
    "lineNumber": 118,
    "issueType": "Token",
    "secretValue": "ghp_yaml_github_token_vwx901"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 46,
    "lineNumber": 119,
    "issueType": "Secret",
    "secretValue": "yaml_github_webhook_yz234"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 47,
    "lineNumber": 122,
    "issueType": "Secret",
    "secretValue": "yaml_slack_webhook_abc567"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 48,
    "lineNumber": 123,
    "issueType": "Token",
    "secretValue": "xoxb-yaml-slack-bot-token-def890"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 49,
    "lineNumber": 126,
    "issueType": "Secret",
    "secretValue": "yaml_csrf_secret_ghi123"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 50,
    "lineNumber": 127,
    "issueType": "Token",
    "secretValue": "yaml_api_security_jkl456"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 51,
    "lineNumber": 128,
    "issueType": "Token",
    "secretValue": "yaml_session_token_mno789"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 52,
    "lineNumber": 130,
    "issueType": "Secret",
    "secretValue": "yaml_inline_secret_pqr012"
  },
  {
    "filePath": "secrets.yml",
    "issueNumber": 53,
    "lineNumber": 130,
    "issueType": "Token",
    "secretValue": "yaml_inline_comment_secret_stu345"
  }
]
